### ISSUE TITLE
Defeating a monster means the next person in initiative order gets skipped

### ISSUE THREAD

**Issue by mpirnat** [mpirnat/dndme#34]
This meant that Pip lost her turn when the thing before her got moved from active to defeated. THIS SUCKS AND LET US NOT HAVE IT HAPPEN AGAIN.

**Comment by mpirnat** [comment]
This happens when multiple combatants share the same initiative value and one of them defeats another that had already had a turn while there are still one or more combatants that have not had their turns yet.

For example, if we have A, B, and C, all acting in initiative 20, and on B's turn, A is defeated, advancing to the next combatant should yield C but instead yields B -- skipping C's turn!

Here's an example where this is recreated:

```
Beginning combat with: 
1: goblin-01/e027, goblin-02/bdb2, goblin-03/828b, goblin-04/e6ce, goblin-05/b47d

> next
Round: 1 Initiative: 1 Name: goblin-01/e027

> next
Round: 1 Initiative: 1 Name: goblin-02/bdb2

> defeat goblin-01/e027
Defeated goblin-01/e027

> next
Round: 1 Initiative: 1 Name: goblin-04/e6ce

> next
Round: 1 Initiative: 1 Name: goblin-05/b47d

> next
Round: 2 Initiative: 1 Name: goblin-02/bdb2
```

**Comment by mpirnat** [comment]
Thought I fixed it with 5690f7f...

```
Beginning combat with: 
1: goblin-01/da43, goblin-02/d0a1, goblin-03/5eca, goblin-04/c0a5

> next
Round: 1 Initiative: 1 Name: goblin-01/da43

> next
Round: 1 Initiative: 1 Name: goblin-02/d0a1

> defeat goblin-01/da43
Defeated goblin-01/da43

> next
Round: 1 Initiative: 1 Name: goblin-03/5eca

> next
Round: 1 Initiative: 1 Name: goblin-04/c0a5

> next
Round: 2 Initiative: 1 Name: goblin-02/d0a1
```

But it turns out I introduced a bug where defeated combatants still get a turn:

```
Beginning combat with: 
1: goblin-01/da43, goblin-02/d0a1, goblin-03/5eca, goblin-04/c0a5

> next
Round: 1 Initiative: 1 Name: goblin-01/da43

> next
Round: 1 Initiative: 1 Name: goblin-02/d0a1

> defeat goblin-01/da43
Defeated goblin-01/da43

> next
Round: 1 Initiative: 1 Name: goblin-03/5eca

> next
Round: 1 Initiative: 1 Name: goblin-04/c0a5

> next
Round: 2 Initiative: 1 Name: goblin-02/d0a1

> defeat goblin-03/5eca
Defeated goblin-03/5eca

> next
Round: 2 Initiative: 1 Name: goblin-03/5eca
```

But then I fixed THAT too (and verified that the original fix still worked):

```
Beginning combat with: 
1: goblin-01/f078, goblin-02/7e7b, goblin-03/1c71, goblin-04/1169

> next
Round: 1 Initiative: 1 Name: goblin-01/f078

> defeat goblin-02/7e7b
Defeated goblin-02/7e7b

> next
Round: 1 Initiative: 1 Name: goblin-03/1c71

> next
Round: 1 Initiative: 1 Name: goblin-04/1169

> next
Round: 2 Initiative: 1 Name: goblin-01/f078

> next
Round: 2 Initiative: 1 Name: goblin-03/1c71

> defeat goblin-01/f078
Defeated goblin-01/f078

> next
Round: 2 Initiative: 1 Name: goblin-04/1169

> next
Round: 3 Initiative: 1 Name: goblin-03/1c71
```
